lib/tree.rb

```diff
@@ -16,40 +16,87 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(log n)
+  # Space Complexity: O(n) - call stack
   def add(key, value)
```

Comment:
👍 The time & space complexities are **both** O(log n) if the tree is balanced and O(n) if unbalanced.